### PR TITLE
Add some standalone tests for runguard.

### PR DESCRIPTION
**Pull request by meisterT**
This is not only useful in general for quicker testing of runguard, but will help to gain some confidence in runguard changes as we plan to do with
https://docs.google.com/document/d/1WZRwdvJUamsczYC7CpP3ZIBU8xG6wNqYqrNJf7osxYs/edit

### CODE REVIEW

File: gitlab/integration.sh [hunk 1]
```diff
@@ -130,10 +127,21 @@ set -e
 if [ $PIN_JUDGEDAEMON -eq 1 ]; then
     PINNING="-n 0"
 fi
+section_end more_setup
+
+section_start runguard_tests "Running isolated runguard tests"
```
Comment (vmcj) [inline gitlab/integration.sh:132]: You're extending the test which takes the longest, can we move this to a dedicated job? Or is the extra time neglectable?

File: judge/runguard_test/runguard_test.sh [hunk 2]
```diff
@@ -0,0 +1,289 @@
+#!/bin/bash
```
Comment (vmcj) [inline judge/runguard_test/runguard_test.sh:1]: I assume you're aware that the other tests use `bats` but decided that this was better?

File: gitlab/integration.sh (hunk 1 above)
Comment (meisterT) [reply gitlab/integration.sh:132]: It takes ~32s on CI, out of ~13mins. I would say it's not worth adding the complication of a dedicated job (until we can enable it on github actions).

File: judge/runguard_test/runguard_test.sh (hunk 2 above)
Comment (meisterT) [reply judge/runguard_test/runguard_test.sh:1]: Yeah, my test is allows for more flexibility that I think was necessary here.

**Review by vmcj — APPROVED** [review]
LGTM but I'm not an expert on runguard.

**Pull request merged**